pkg/actions/service_test.go

```diff
@@ -38,57 +40,86 @@ func TestServiceRun(t *testing.T) {
-			t.Errorf("Webhook post RepositoryID=%s, expected=%s", eventInfo.RepositoryID, record.RepositoryID)
+		if event.SourceRef != record.SourceRef.String() {
+			t.Errorf("Webhook post SourceRef=%s, expected=%s", event.SourceRef, record.SourceRef)
 		}
-		if eventInfo.BranchID != record.BranchID.String() {
-			t.Errorf("Webhook post BranchID=%s, expected=%s", eventInfo.BranchID, record.BranchID)
+		if event.CommitMessage != record.Commit.Message {
+			t.Errorf("Webhook post CommitMessage=%s, expected=%s", event.CommitMessage, record.Commit.Message)
 		}
-		if eventInfo.SourceRef != record.SourceRef.String() {
-			t.Errorf("Webhook post SourceRef=%s, expected=%s", eventInfo.SourceRef, record.SourceRef)
+		if event.Committer != record.Commit.Committer {
+			t.Errorf("Webhook post Committer=%s, expected=%s", event.Committer, record.Commit.Committer)
 		}
-		if eventInfo.CommitMessage != record.Commit.Message {
-			t.Errorf("Webhook post CommitMessage=%s, expected=%s", eventInfo.CommitMessage, record.Commit.Message)
+		if diff := deep.Equal(event.CommitMetadata, map[string]string(record.Commit.Metadata)); diff != nil {
+			t.Errorf("Webhook post Metadata diff=%s", diff)
 		}
-		if eventInfo.Committer != record.Commit.Committer {
-			t.Errorf("Webhook post Committer=%s, expected=%s", eventInfo.Committer, record.Commit.Committer)
+	}
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer func() { _ = r.Body.Close() }()
+		data, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			t.Error("Failed to read webhook post data", err)
+			return
 		}
-		if diff := deep.Equal(eventInfo.CommitMetadata, map[string]string(record.Commit.Metadata)); diff != nil {
-			t.Errorf("Webhook post Metadata diff=%s", diff)
+
+		if r.URL.Path == "/webhook" {
+			queryParams := map[string][]string(r.URL.Query())
+			require.Len(t, queryParams["prefix"], 1)
+			require.Equal(t, "public/", queryParams["prefix"][0])
+			require.Len(t, queryParams["disallow"], 2)
+			require.Equal(t, "user_", queryParams["disallow"][0])
+			require.Equal(t, "private_", queryParams["disallow"][1])
+
+			var eventInfo actions.EventInfo
+			err = json.Unmarshal(data, &eventInfo)
+			if err != nil {
+				t.Error("Failed to unmarshal webhook data", err)
+				return
+			}
+
+			checkEvent(eventInfo, webhookID)
+		} else if r.URL.Path == "/airflow/api/v1/dags/some_dag_id/dagRuns" {
+			var req actions.DagRunReq
+			require.NoError(t, json.Unmarshal(data, &req))
+			require.True(t, strings.HasPrefix(*req.DagRunID, "lakeFS_hook_"+airflowHookID))
+			require.Equal(t, req.Conf["some"], "additional_conf")
+
+			username, pass, ok := r.BasicAuth()
+			require.True(t, ok)
+			require.Equal(t, "some_username", username)
+			require.Equal(t, "some_password", pass)
+			rawEvent, ok := req.Conf["lakeFS_event"]
+			require.True(t, ok, "missing lakeFS event")
+
+			var event actions.EventInfo
+			rawBytes, err := base64.StdEncoding.DecodeString(rawEvent.(string))
```

Comment:
encode json.RawMessage will prevent decode base64 and the actual implementation can parse the data as part of the original message

Reply:
Done, thanks!